packages/ember-application/tests/system/dependency_injection/default_resolver_test.js

```diff
@@ -170,3 +176,34 @@ QUnit.test("lookup description", function() {
   equal(registry.describe('model:foo'), 'App.Foo', "models don't get appended at the end");
 
 });
+
+QUnit.test("validating resolved objects", function() {
+  let types = ['route', 'component', 'view', 'service'];
+
+  // Valid setup
+  application.FooRoute = Route.extend();
+  application.FooComponent = Component.extend();
+  application.FooView = View.extend();
+  application.FooService = Service.extend();
+
+  forEach(types, function(type) {
+    // No errors when resolving correct object types
+    registry.resolve(`${type}:foo`);
+
+    // Unregister to clear cache
+    registry.unregister(`${type}:foo`);
+  });
+
+  // Invalid setup
+  application.FooRoute = Component.extend();
+  application.FooComponent = View.extend();
+  application.FooView = Service.extend();
+  application.FooService = Route.extend();
+
+  forEach(types, function(type) {
+    let matcher = new RegExp(`to resolve to an Ember.${capitalize(type)}`);
+    throws(function() {
```

Comment:
Ember.assert statements are stripped from production builds, so the prod Travis run is failing (since nothing is thrown). We need to use 'expectAssertion' here, and that helper function will handle the prod case for you.